app/views/agents/_table.html.erb

```diff
@@ -14,7 +14,7 @@
     <% @agents.each do |agent| %>
       <tr>
         <td class='<%= "agent-unavailable" if agent.unavailable? %>'>
-          <%= link_to agent.name, agent_path(agent) %>
+          <%= link_to agent.name, agent_path(agent, return: (defined?(return_to) && return_to) || request.path) %>
```

Comment:
I just googled and read `defined?()` would work reliably for view locals in Rails 4.2, where you would have previously used `local_assigns.has_key?`.  Nice to know!